pkg/operator/controller/controller.go

```diff
@@ -165,7 +168,13 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 			} else {
 				// Handle everything else.
 				if err := r.ensureDNS(dns); err != nil {
-					errs = append(errs, fmt.Errorf("failed to ensure dns %s: %v", dns.Name, err))
+					switch e := err.(type) {
+					case retry.Error:
+						// Got retryable error; requeue
+						return reconcile.Result{RequeueAfter: e.After()}, nil
+					default:
+						errs = append(errs, fmt.Errorf("failed to ensure dns %s: %v", dns.Name, err))
+					}
```

Comment:
Would it be better to leave this logic unchanged and add the retry logic at the end of the `Reconcile` method?  Something like this:
```go
				if err := r.ensureDNS(dns); err != nil {
￼					errs = append(errs, fmt.Errorf("failed to ensure dns %s: %v", dns.Name, err))
				} else if err := r.ensureExternalNameForOpenshiftService(); err != nil {
					errs = append(errs, fmt.Errorf("failed to ensure external name for openshift service: %v", err))
				}
			}
		}

	if err := retry.NewMaybeRetryableAggregate(errs); err != nil {
		switch e := err.(type) {
		case retryable.Error:
			log.Error(e, "got retryable error; requeueing", "after", e.After())
			return reconcile.Result{RequeueAfter: e.After()}, nil
		default:
			// Log in case of errors as the controller's logs get eaten.
			logrus.Errorf("failed to reconcile request %s: %v", request, err)
			return reconcile.Result{}, err
		}
	}
	return reconcile.Result{}, nil
```￼

Reply:
Thanks, that's better!  Though I'll still need to `utilerrors.NewAggregate(errs)` in the default case instead of just `err`.  Done.